lib/tslint/6.0.x/tslint.json

```diff
@@ -0,0 +1,225 @@
+    "align": [false, "arguments", "statements", "parameters", "members", "elements"],
+    "callable-types": true,
+    "class-name": true,
+    "comment-format": [true, "check-space"],
+    "file-header": false,
+    "encoding": false,
+    "interface-name": [false, "never-prefix"],
+    "interface-over-type-literal": true,
+    "jsdoc-format": [true, "check-multiline-start"],
+    "no-angle-bracket-type-assertion": false,
+    "no-parameter-properties": false,
+    "no-redundant-jsdoc": true,
+    "no-reference-import": true,
+    "no-unnecessary-callback-wrapper": false,
+    "object-literal-shorthand": false,
+    "one-variable-per-declaration": true,
+    "switch-final-break": false,
+    "variable-name": [
+      true,
+      "check-format",
+      "require-const-for-all-caps",
+      "allow-leading-underscore",
+      "allow-trailing-underscore",
+      "ban-keywords"
+    ],
+    "whitespace": [
+      false,
+      "check-branch",
+      "check-decl",
+      "check-operator",
+      "check-module",
+      "check-separator",
+      "check-rest-spread",
+      "check-type",
+      "check-typecast",
+      "check-type-operator",
+      "check-preblock"
+    ],
+    "prefer-const": [
+      true,
+      {
+        "destructuring": "all"
+      }
+    ],
+    "no-magic-numbers": false,
+    "import-blacklist": [true, "@angular/material", "lodash", "rxjs/internal", "rxjs/index", "rxjs/Rx", "lodash-es"],
+    "ordered-imports": false,
+    "max-classes-per-file": false, // [true, 1],
+    "prefer-for-of": true,
+    "array-type": false, // [true, "array"]
+    "arrow-return-shorthand": true,
+    "binary-expression-operand-order": false,
+    "no-unnecessary-initializer": true,
+    "no-misused-new": true,
+    "prefer-method-signature": false,
+    "prefer-switch": [
+      true,
+      {
+        "min-cases": 3
+      }
+    ],
+    "prefer-function-over-method": false,
+    "prefer-while": true,
+    "no-import-side-effect": [
+      true,
+      {
+        "ignore-module": "(reflect-metadata|core-js|ngrx-store|angular|moment|polyfills|vendor|typings|\\.css|\\.pcss)"
+      }
+    ],
+    "no-non-null-assertion": true,
+    "no-parameter-reassignment": false,
+    "adjacent-overload-signatures": true,
+    "ban-comma-operator": false,
+    "ban-types": false,
+    "no-duplicate-super": true,
+    "no-duplicate-switch-case": true,
+    "newline-before-return": false,
+    "newline-per-chained-call": false,
+    "file-name-casing": [true, { "public_api.ts": "snake-case", ".*": "kebab-case" }],
+    "ban-ts-ignore": true,
+    "function-constructor": true,
+    "unnecessary-constructor": [true, { "check-super-calls": true }],
+    "no-tautology-expression": true,
+    "static-this": true,
+    "unnecessary-else": [true, { "allow-else-if": true }],
+    "no-async-without-await": true,
+    "invalid-void": true,
+    // TSLint rules that were previously on tslint:recommended prior to v6.0
+    //"arrow-parens": true,
```

Comment:
@SuperITMan @nicanac I'm not sure what to do with these rules. They were removed from `tslint:recommended` in TSLint v6.0 in https://github.com/palantir/tslint/pull/4871/files and https://github.com/palantir/tslint/pull/4312/files

Normally that shouldn't affect but, in fact in Stark we extend our configs from `tslint:latest` which actually extends from `tslint:recommended` (see https://github.com/palantir/tslint/blob/master/src/configs/latest.ts) therefore I was thinking on adding them here... but we should also consider that these rules are disabled by `tslint-config-prettier` (which we use in Stark :p).

So.. what do you think we should do here? Should we add the rules here anyway to ensure that any project using `code-style` without `Prettier` keeps the same rules? or we just don't care so we don't add them here so any impacted project would need to add those by themselves?

Reply:
I've created PR https://github.com/NationalBankBelgium/stark/pull/1655 where you can see all the adaptations needed in Stark based on the new TSLint rules/options enabled from 5.10 to 6.0.

Reply:
The rules listed in here are the ones that were removed from the `tslint:recommended` and in fact those rules are disabled by `tslint-config-prettier` in Stark, hence my question: 
> So.. what do you think we should do here? Should we add the rules here anyway to ensure that any project using code-style without Prettier keeps the same rules? or we just don't care so we don't add them here so any impacted project would need to add those by themselves?

About the `whitespace` rule, it is disabled in our configs anyway ;)

Reply:
I would say "we don't care". 
I think we can keep those rules in comments as you did, like this it would be easier for the dev to find the previous rules. 

Maybe we could add an explanation in the README of tsconfig about the fact that we recommend to use `tslint-config-prettier` and `prettier` in addition to the config we provide 😊 

What do you think?